plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClientModule.java

```diff
@@ -68,6 +68,7 @@ public static Properties getConnectionProperties(MySqlConfig mySqlConfig)
         connectionProperties.setProperty("useUnicode", "true");
         connectionProperties.setProperty("characterEncoding", "utf8");
         connectionProperties.setProperty("tinyInt1isBit", "false");
+        connectionProperties.setProperty("rewriteBatchedStatements", "true");
```

Comment:
i do see this flag mentioned many times in release notes (https://dev.mysql.com/doc/search/?q=rewriteBatchedStatements&d=188&p=1), but couldn't find documentation.

I like the change. 
Is there a docs link? Do we need a kill-switch? (i hope we do not cc @hashhar @ebyhr )

Reply:
https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-connp-props-performance-extensions.html#cj-conn-prop_rewriteBatchedStatements

Reply:
I'd prefer to not add kill switch and it seems that this mechanism is safe to use in our scenarios (also it kicks in where several conditions are met).

Reply:
from the page linked above

> Notice ...  that if you don't specify stream lengths when using PreparedStatement.set*Stream(), the driver won't be able to determine the optimum number of parameters per batch and you might receive an error from the driver that the resultant packet is too large

Reply:
> We are not using set*Stream :)

Today.

and if we start using them, and problem is data-size dependent, we won't know about a potential problem.

Reply:
I guess ultimately we may want to expose the batchWrite enable/ batchSize as config parameters, (and document them) for full flexibility? Enabled by default for max performance, but configurable if someone has a use-case which demands it (and it would fulfill the 'kill switch' question, because consumers could set `catalog.batchWrite.enabled=false`, etc)?

Reply:
(I can imagine a scenario where a (sub-optimal, but working) insert starts to fail because the batch is too big for the maximum statement size (maybe `max_allowed_packet` is too big to fail from an insert - but if people are working with blobs, etc,) batching could break in a lot of places to not have some sort of "escape hatch" - and probably better to add it + test it before we need to 'test it as a workaround')...

Reply:
The driver handles the rewriting or not based on actual statements and parameters used. It only fails to do so if you use `setStream`.

We can add a toggle to disable this feature if needed. There is no "batch size" here.

Reply:
@hashhar - I wasn't referring to setStream, but I see that (as you say) the driver handles the rewrites based on parameters:
```
(from https://stackoverflow.com/questions/26307760/mysql-and-jdbc-with-rewritebatchedstatements-true )
Q:  does JDBC knows the value assigned to max_allowed_packet and therefore make the packet smaller than the defined value for max_allowed_packet ... ?

Yes. If you enable the MySQL general log and check it you will see that MySQL Connector/J inspects a bunch of variables when it connects, one of which is max_allowed_packet. You can also set a small max_allowed_packet value and verify that JDBC splits a batch into several multi-row INSERT statements if a single such statement for the whole batch would exceed max_allowed_packet
```
👍

Reply:
Before / instead of adding a toggle, we should first see what logic can be applied automatically.
For example, if a user wants to configure batch size, we should ask ourselves, what's the use-case and whether we can detect this automatically. Doing this is more complicating that just throwing a toggle, but helps everyone. A toggle helps only those few people who'd tinker with it.